# Work around issues with XSyncCreateFence() in the sparc version of
# the lucid interface. See:
#
#    https://debbugs.gnu.org/cgi/bugreport.cgi?bug=69762
#
--- emacs-29.3.orig/src/xterm.c	2024-01-06 05:56:31.000000000 -0700
+++ emacs-29.3/src/xterm.c	2024-03-25 13:29:17.252101887 -0600
@@ -7061,6 +7061,7 @@
 	  && dpyinfo->xsync_minor < 1))
     return;
 
+  x_ignore_errors_for_next_request (dpyinfo);
   output->sync_fences[0]
     = XSyncCreateFence (FRAME_X_DISPLAY (f),
 			/* The drawable given below is only used to
@@ -7072,6 +7073,7 @@
     = XSyncCreateFence (FRAME_X_DISPLAY (f),
 			FRAME_X_WINDOW (f),
 			False);
+  x_stop_ignoring_errors (dpyinfo);
 
   XChangeProperty (FRAME_X_DISPLAY (f), FRAME_OUTER_WINDOW (f),
 		   dpyinfo->Xatom_net_wm_sync_fences, XA_CARDINAL,
